string/string.h

```diff
@@ -0,0 +1,267 @@
+/**
+ * Library for embedded strings
+ * 
+ * @author Aaron Chan
+ */
+
+#ifndef STRING_H
+#define STRING_H
+
+#include <cstddef>
+
+const char NULL_TERMINATOR = '\0';
+
+template<const size_t size>
+class String {
+public:
+    int len = 0;
+    char string[size] = {};
+
+    /**
+     * String constructor
+     * @param buffer
+     */
+    explicit String(char const *buffer) {
+        int i; // Reduce errors from using len as counter
+
+        for (i = 0; i < size - 1; i++) {
+            this->string[i] = buffer[i];
+
+            if (this->string[i] == NULL_TERMINATOR) break;
+        }
+
+        // Developers should still add a null terminator in their buffers, especially if len < buff_size
+        if (this->string[i] != NULL_TERMINATOR) {
+            this->string[i] = '\0';
+        }
+
+        this->len = i - 1;
+    }
+
+    /**************************************
+     * Element Access
+     **************************************/
+
+    /**
+     * Retrieves first value in string
+     * @return First Element in String
+     */
+    char front() {
+        return this->string[0];
+    }
+
+    /**
+     * Retrieves last value in string
+     * @return Last Element in String
+     */
+    char back() {
+        return this->string[len];
+    }
+
+    /**
+     * Retrieves value at specified index
+     * @param index to retrieve element at
+     * @return Element at index or nullptr if invalid
+     */
+    char at(int index) {
```

Comment:
it'd be cool if you overloaded the [] operator to what at does. maybe make it return a reference so you can set it with [] as well?